src/node/chainstatemanager_args.cpp

```diff
@@ -32,13 +32,20 @@ util::Result<void> ApplyArgsManOptions(const ArgsManager& args, ChainstateManage
     if (auto value{args.GetBoolArg("-checkpoints")}) opts.checkpoints_enabled = *value;
 
     if (auto value{args.GetArg("-minimumchainwork")}) {
-        if (!IsHexNumber(*value)) {
+        if (auto sanitized_hex{TrySanitizeHexNumber(*value, /*result_size=*/64)}) {
+            opts.minimum_chain_work = UintToArith256(*uint256::FromHex(*sanitized_hex));
+        } else {
             return util::Error{strprintf(Untranslated("Invalid non-hex (%s) minimum chain work value specified"), *value)};
         }
-        opts.minimum_chain_work = UintToArith256(uint256S(*value));
     }
 
-    if (auto value{args.GetArg("-assumevalid")}) opts.assumed_valid_block = uint256S(*value);
+    if (auto value{args.GetArg("-assumevalid")}; value && *value != "0") {
```

Comment:
shouldn't we still set the `assumed_valid_block` value when it's `0`, given that `defaultAssumeValid` isn't `0`?

Reply:
Ugh, you're right, thanks a lot for catching this. This approach would've silently ignored `-noassumevalid` and `-assumevalid=0` by using the `defaultAssumeValid` value instead. I think it's not great that this bug didn't cause any tests to fail. `feature_assumevalid.py` doesn't catch this because `defaultAssumeValid` on regtest [is null](https://github.com/bitcoin/bitcoin/blob/43740f4971f45cd5499470b6a085b3ecd8b96d28/src/kernel/chainparams.cpp#L452) (and I don't think there's a way around this), and we don't seem to have any unittesting on applying args to options.

I've added e9dd47c66426c04fc6f0780d67ba0c05f26f9920 as a first commit (fails on b38a259d1e2749876d4e3a15f218b4f257049320) to add unittesting on `-assumevalid` and `-minimumchainwork`.